### DISCUSSION TITLE
Gestion de mot de passe

### DISCUSSION THREAD

**Discussion by JYK-WIT** [witsa/synapps#539] [Ideas]
Bonjour,
il serait pas mal d'avoir une scène ou modale pour la gestion des mots de passe. 
Un administrateur pourrait créer/supprimer/modifier un utilisateur avec un niveau d'accès souhaité et un mot de passe générique.
L'utilisateur pourrait seulement modifier son mot de passe 
Bien sûr avec le contrôle nécessaire pour avoir un MdP durci.
et rappel pour le changement régulier si nécessaire
JYK

**Comment by manufitoussiwit** [comment]
Salut! Effectivement, j'ai regardé si on pouvait faire ça avec Synapps mais malheureusement, actuellement, un utilisateur exploitant ne peut pas accéder aux nœuds de son compte utilisateur. C'est dommage. Il peut même pas afficher dans iFrame l'interface du REDY qui fait ça.
Par contre, je me suis rendu compte qu'un exploitant ne peux pas non plus faire ça depuis l'interface du REDY ! 
Je me trompe ?

**Comment by WIToliv** [comment]
Salut,
Une question, mon client me demande de limiter les actions à certains utilisateurs.
Avant quand on créait un utilisateur "Invité" dans le REDY, il pouvait voir des pages mais rien modifier dans les synoptiques.
Là, avec un compte invité, limité sur le REDY, il peut tout faire.
Comment on gère cela?
Merci
OV

**Reply by manufitoussiwit** [reply]
> Salut, Une question, mon client me demande de limiter les actions à certains utilisateurs. Avant quand on créait un utilisateur "Invité" dans le REDY, il pouvait voir des pages mais rien modifier dans les synoptiques. Là, avec un compte invité, limité sur le REDY, il peut tout faire. Comment on gère cela? Merci OV

Salut!

Les acteurs détails de reflet suivent les droits utilisateurs. Un invité ne peut pas commander un reflet par défaut.

Avec les fournisseurs de variable par contre, il faut ajouter une liaison sur l'activité du bouton/saisie/checkbox/... vers le champ `Ecriture ?` qui permet de savoir s'il l'utilisateur peut écrire dans la variable : 

![image](https://user-images.githubusercontent.com/35595723/194266074-77960992-7fea-434a-b1e6-4cab5a9c5227.png)

Si on ne le fait pas, de toute manière, la commande retourne une erreur (heureusement ;-) )

**Reply by manufitoussiwit** [reply]
Il ne faut pas hésiter à utiliser plusieurs hôtes avec des utilisateurs à niveau/groupe différent pour tester et basculer de l'un à l'autre.
Par exemple : 
## Admin : 

![image](https://user-images.githubusercontent.com/35595723/194267792-7ed09f64-7101-4b46-bef8-08f746fda547.png)


## Invité :
![image](https://user-images.githubusercontent.com/35595723/194267701-f83133eb-b6b1-4ece-be77-ce086ef12f92.png)


## Et pour basculer vite fait :
![image](https://user-images.githubusercontent.com/35595723/194267550-e3cbbd7d-4b8f-4c8e-befa-5d1579ef1267.png)
